app/questionnaire/placeholder_parser.py

```diff
@@ -28,6 +28,7 @@ def __init__(
         answer_store: AnswerStore,
         list_store: ListStore,
         metadata: ImmutableDict,
+        response_metadata: dict,
```

Comment:
Same with placeholders renderer/parser. We should change to `Mapping`.

Reply:
Changed to `Mapping`, thanks!